### PR TITLE
Configurable assets stream extensions

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | https://github.com/mautic/user-documentation/pull/192
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Follow up for https://github.com/mautic/mautic/pull/11043
Some extensions have to be download everytime, for example ICS. 
Then we've added list of extensions to display in browser by default. Others are download as before

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create PDF asset
3. Check If download link open the PDF
4. Create CSV asset
5. Check If download link download the CSV

### CODE REVIEW

File: app/bundles/AssetBundle/Controller/AssetController.php [hunk 1]
```diff
@@ -229,7 +229,12 @@ public function previewAction($objectId)
         }
 
         $download = $this->request->query->get('download', 0);
-        $stream   = $this->request->query->get('stream', 0);
+
+        // Display the file directly in the browser just for selected extensions
+        $stream   = $this->request->query->get('stream', in_array($activeAsset->getExtension(), $this->coreParametersHelper->get('streamed_extensions')));
+
+        $fileStorageEvent = new StorageAssetFileEvent($activeAsset->getFilePath());
```
Comment (mollux) [inline app/bundles/AssetBundle/Controller/AssetController.php]: this class isn't referenced in a `use` statement, so this will fail

**Review by mollux — CHANGES_REQUESTED** [review]

**Review by volha-pivavarchyk — DISMISSED** [review]
Works well. Assets are streamed as expected

**Comment by kuzmany** [comment]
I've rebased it based on 5.x

**Review by RCheesley — DISMISSED** [review]
Thanks for this, I recall having this discussion at the time the PR was created!

This PR will need documentation to be added to the configuration section, pinging @favour-chibueze because of course we need to add more to that epic configuration PR for the new docs! 😅

**Comment by kuzmany** [comment]
@RCheesley in fact, this do not have configuration.  It just allow for any company change list of stream able extensions by parameter.

**Comment by RCheesley** [comment]
Looks like this PR needs some attention - conflicts and failing tests to address.

**Comment by kuzmany** [comment]
Conflicts resolved and documentation added.

**Review by RCheesley — APPROVED** [review]
Works as explained - PDF is opened in a new tab, CSV is downloaded.

**Review by escopecz — APPROVED** [review]
👍 the code makes sense

**Pull request merged**